tests/e2e/concurrent-e2e/concurrent_test.go

```diff
@@ -0,0 +1,369 @@
+						Duration: &pipelineSpecSourceDuration,
+					},
+				},
+				Scale: numaflowv1.Scale{Min: &fourPods, Max: &fivePods, ZeroReplicaSleepSeconds: &zeroReplicaSleepSec},
+			},
+			{
+				Name: "cat",
+				UDF: &numaflowv1.UDF{
+					Builtin: &numaflowv1.Function{
+						Name: "cat",
+					},
+				},
+				Scale: numaflowv1.Scale{Min: &onePod, Max: &onePod, ZeroReplicaSleepSeconds: &zeroReplicaSleepSec},
+			},
+			{
+				Name: "out",
+				Sink: &numaflowv1.Sink{
+					AbstractSink: numaflowv1.AbstractSink{
+						Log: &numaflowv1.Log{},
+					},
+				},
+				Scale: numaflowv1.Scale{Min: &onePod, Max: &onePod, ZeroReplicaSleepSeconds: &zeroReplicaSleepSec},
+			},
+		},
+		Edges: []numaflowv1.Edge{
+			{
+				From: sourceVertexName,
+				To:   "cat",
+			},
+			{
+				From: "cat",
+				To:   "out",
+			},
+		},
+	}
+
+	monoVertexScaleMin = int32(4)
+	monoVertexScaleMax = int32(5)
+
+	initialMonoVertexSpec = numaflowv1.MonoVertexSpec{
+		Scale: numaflowv1.Scale{
+			Min: &monoVertexScaleMin,
+			Max: &monoVertexScaleMax,
+		},
+		Source: &numaflowv1.Source{
+			UDSource: &numaflowv1.UDSource{
+				Container: &numaflowv1.Container{
+					Image: "quay.io/numaio/numaflow-go/source-simple-source:stable",
+				},
+			},
+		},
+		Sink: &numaflowv1.Sink{
+			AbstractSink: numaflowv1.AbstractSink{
+				Blackhole: &numaflowv1.Blackhole{},
+			},
+		},
+	}
+
+	updatedMonoVertexSpec = numaflowv1.MonoVertexSpec{
+		Scale: numaflowv1.Scale{
+			Min: &monoVertexScaleMin,
+			Max: &monoVertexScaleMax,
+		},
+		Source: &numaflowv1.Source{
+			UDSource: &numaflowv1.UDSource{
+				Container: &numaflowv1.Container{
+					Image: "quay.io/numaio/numaflow-java/source-simple-source:stable",
+				},
+			},
+		},
+		Sink: &numaflowv1.Sink{
+			AbstractSink: numaflowv1.AbstractSink{
+				Blackhole: &numaflowv1.Blackhole{},
+			},
+		},
+	}
+
+	volSize, _            = apiresource.ParseQuantity("10Mi")
+	initialISBServiceSpec = numaflowv1.InterStepBufferServiceSpec{
+		Redis: nil,
+		JetStream: &numaflowv1.JetStreamBufferService{
+			Version: InitialJetstreamVersion,
+			Persistence: &numaflowv1.PersistenceStrategy{
+				VolumeSize: &volSize,
+			},
+		},
+	}
+
+	dataLossISBServiceSpec = numaflowv1.InterStepBufferServiceSpec{
+		Redis: nil,
+		JetStream: &numaflowv1.JetStreamBufferService{
+			Version: UpdatedJetstreamVersion,
+			Persistence: &numaflowv1.PersistenceStrategy{
+				VolumeSize: &volSize,
+			},
+		},
+	}
+
+	updatedMemLimit, _ = apiresource.ParseQuantity("2Gi")
+	// directApplyISBServiceSpec = numaflowv1.InterStepBufferServiceSpec{
+	// 	Redis: nil,
+	// 	JetStream: &numaflowv1.JetStreamBufferService{
+	// 		Version: initialJetstreamVersion,
+	// 		Persistence: &numaflowv1.PersistenceStrategy{
+	// 			VolumeSize: &volSize,
+	// 		},
+	// 		ContainerTemplate: &numaflowv1.ContainerTemplate{
+	// 			Resources: v1.ResourceRequirements{
+	// 				Limits: v1.ResourceList{v1.ResourceMemory: updatedMemLimit},
+	// 			},
+	// 		},
+	// 	},
+	// }
```

Comment:
is this still here because you may use it in a future test? or should we clean it up?